api/v1alpha1/volumereplicationgroup_types.go

```diff
@@ -18,53 +18,38 @@ package v1alpha1
-	ForcePromote VolumeTakeoverControlType = "ForcePromote"
-)
-
-// VolumeReplicationGroupSpec defines the desired state of VolumeReplicationGroup
+// VolumeReplicationGroup (VRG) spec declares the desired replication class
+// and replication state of all the PVCs owned by a given application, which
+// are identified via the given PVC label selector.  For every PVC created
+// by the application, the VRG operator does the following:
+// 	- Create a VolumeReplication (VR) CR to enable storage level replication
+// 	  of volume data and set the volume state to primary, secondary, etc.
+//  - Take the corresponding PV metadata in Kubernetes etcd and deposit it in
+//    the S3 store.  The url, access key and access id required to access the
+//    S3 store is specified via environment variables of the VRG operator POD,
+//    which is obtained from a secret resource.
+//  - Manage the lifecycle of VR CR and S3 data to match the CUD operations on
+//    VRG CR.
 type VolumeReplicationGroupSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	// Name of the application to replicate
-	ApplicationName string `json:"applicationName"`
-
-	// Application label that is used to identify its PVs
-	ApplicationLabels metav1.LabelSelector `json:"applicationLabels"`
-
-	// Cluster ID in ClusterPeers that has best storage performance affinity for the application
-	AffinedCluster string `json:"affinedCluster"`
-
-	// Desired cluster that should takeover the application from current active cluster.
-	// May be set either to:
-	// - secondary cluster ID (to migrate application or takeover application in case of a disaster)
-	// - nil (if application is not active in the affined cluster, takeback to affined cluster)
-	// +optional
-	DesiredCluster string `json:"desiredCluster,omitempty"`
-
-	// Volume Takeover Control: ForcePromote
-	// +optional
-	VolumeTakeoverControl VolumeTakeoverControlType `json:"volumeTakeoverControl,omitempty"`
+	// Label selector to identify all the PVCs that are in this group
+	// This operator will create and needs to be replicated
+	PVCLabelSelector metav1.LabelSelector `json:"pvcLabelSelector"`
 
-	// List of ClusterPeers
-	// For Metro DR only: a single ClusterPeers
-	// For WAN DR only: one or more ClusterPeers
-	// For MetroDR and WAN DR: one Metro DR ClusterPeer and one or more WAN DR ClusterPeers.
-	ClusterPeersList []string `json:"clusterPeersList"`
+	// ReplicationClass of all volumes in this replication group that is
+	// passed to VolumeReplication CR
+	VolumeReplicationClass string `json:"volumeReplicationClass"`
 
-	// WAN DR RPO goal in seconds
-	AsyncRPOGoalSeconds int64 `json:"asyncRPOGoalSeconds,omitempty"`
+	// Desired state of all volumes [primary or secondary] in this replication group that is
+	// passed to VolumeReplication CR
+	VolumeState vrv1alpha1.ImageState `json:"volumeState"`
```

Comment:
[kube-storage](https://github.com/kube-storage/volume-replication-operator/pull/47) calls this `replicationState` should we consider the same?

Reply:
I agree; I did suggest it in the issue you opened.  But this change is subject to the above "redundant" prefix discussion.

Reply:
Ok I did not notice that we are using the state as in VolumeReplication from vrv1alpha. We already differ in the states we offer, [Primary|Secondary] and VolRep adds Resync to the mix. So we should hence define our own states and not import them from VolRep.